packages/invoices/src/OutboundShipment/DetailView/modals/ItemBatches.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import {
+  BasicTextInput,
+  Item,
+  StockLine,
+  Table,
+  TableBody,
+  TableCell,
+  TableCellProps,
+  TableContainer,
+  TableHead,
+  TableRow,
+  useFormContext,
+  useFormatDate,
+  useTranslation,
+  Checkbox,
+  Divider,
+  InfoIcon,
+} from '@openmsupply-client/common';
+
+export interface ItemBatchesProps {
+  item: Item | null;
+}
+
+const BatchRow = ({ batch, label }: { batch: StockLine; label: string }) => {
+  const { register } = useFormContext();
+  const t = useTranslation();
+  const d = useFormatDate();
+
+  const stockLineInputProps = register(batch.id, {
+    min: { value: 1, message: t('error.greater-than-zero-required') },
+    pattern: { value: /^[0-9]+$/, message: t('error.number-required') },
+  });
+
+  // TODO format currency correctly
+  return (
+    <TableRow>
+      <BasicCell align="right">{label}</BasicCell>
+      <BasicCell sx={{ width: '88px' }}>
+        <BasicTextInput
+          {...stockLineInputProps}
+          sx={{ height: '32px' }}
+          disabled={batch.availableNumberOfPacks === 0}
+        />
+      </BasicCell>
+      <BasicCell>
+        <Checkbox disabled={batch.availableNumberOfPacks === 0} />
+      </BasicCell>
+      <BasicCell align="right">{batch.totalNumberOfPacks}</BasicCell>
+      <BasicCell align="right">{batch.availableNumberOfPacks}</BasicCell>
+      <BasicCell align="right">{batch.packSize}</BasicCell>
+      <BasicCell>{batch.batch}</BasicCell>
+      <BasicCell>{d(new Date(batch.expiryDate))}</BasicCell>
+      <BasicCell align="right">${batch.costPricePerPack}</BasicCell>
+      <BasicCell align="right">${batch.sellPricePerPack}</BasicCell>
+      <BasicCell>
+        <InfoIcon
+          fontSize="small"
+          sx={{ color: theme => theme.palette.lightGrey }}
+        />
+      </BasicCell>
+    </TableRow>
+  );
+};
+
+const HeaderCell: React.FC<TableCellProps> = ({ children }) => (
+  <BasicCell
+    sx={{
+      color: theme => theme.typography.body1.color,
+      fontWeight: 'bold',
+      padding: '8px',
+    }}
+  >
+    {children}
+  </BasicCell>
+);
+
+const BasicCell: React.FC<TableCellProps> = ({ sx, ...props }) => (
+  <TableCell
+    {...props}
+    sx={{
+      borderBottomWidth: 0,
+      color: theme => theme.palette.darkGrey,
+      fontSize: '12px',
+      padding: '0 8px',
+      whiteSpace: 'nowrap',
+      ...sx,
+    }}
+  />
+);
+
+export const ItemBatches: React.FC<ItemBatchesProps> = ({ item }) => {
+  if (!item) return null;
+
```

Comment:
Would be a good place to put a query later. If there are a lot of items we likely don't want to be getting all the batches also, so we could have a query where the `item.id` is a part of the query key